src/daft-dsl/src/expr.rs

```diff
@@ -576,14 +576,14 @@ impl Expr {
         }
     }
 
-    pub fn to_sql(&self) -> Option<String> {
-        fn to_sql_inner<W: Write>(expr: &Expr, buffer: &mut W) -> io::Result<()> {
+    pub fn to_sql(&self, db_scheme: &str) -> Option<String> {
```

Comment:
This is getting a bit messy now, it may be worth looking into a sql generator package in the near future

Reply:
Oh yeah for sure, that'll be next on my list of todos!